composition/composition.md

```diff
@@ -198,6 +239,96 @@ The total cost of the products in the shopping cart is $213.48
 
 ## Check for Understanding
 
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: multiple-choice
+* id: bcd60833-bf2b-442d-b5b2-05afd630944f
+* title: Composition
+<!-- * points: [1] (optional, the number of points for scoring as a checkpoint) -->
+<!-- * topics: [python, pandas] (optional the topics for analyzing points) -->
+
+##### !question
+
+In the example above, `Product` is the
+
+##### !end-question
+
+##### !options
+
+* Component Class
+* Composite Class
+* Child Class
+* Parent Class
+
+##### !end-options
+
+##### !answer
+
+* Component Class
+
+##### !end-answer
+
+<!-- other optional sections -->
+<!-- !hint - !end-hint (markdown, hidden, students click to view) -->
+<!-- !rubric - !end-rubric (markdown, instructors can see while scoring a checkpoint) -->
+<!-- !explanation - !end-explanation (markdown, students can see after answering correctly) -->
+##### !explanation
+
+An instance of a `ShoppingCart` has many `Product`s.
```

Comment:
consider expanding the explanation a little
```suggestion
An instance of a `ShoppingCart` has many `Product`s. Since instances of `Product` are contained in `ShoppingCart`, `Product` is the component class, while `ShoppingCart` is the composite class.
```